sdk/servicebus/service-bus/src/queueClient.ts

```diff
@@ -46,12 +46,12 @@ export class QueueClient implements Client {
    *
    * @constructor
    * @internal
-   * @param name The Queue name.
+   * @param queueName The Queue name.
    * @param context The connection context to create the QueueClient.
    */
-  constructor(name: string, context: ConnectionContext) {
+  constructor(queueName: string, context: ConnectionContext) {
     throwErrorIfConnectionClosed(context);
-    this.entityPath = name;
+    this.entityPath = queueName.toString();
```

Comment:
Do you think if we should add a check here, if `queueName` is already a string type then set as is, otherwise change to string type  `queueName.toString()`?

Because if it's already a string type then we don't need to call `.toString()` method.

Reply:
I thought about it too, but chose not to do so in favor of brevity. 
cc @bterlson

Reply:
`String(...)` is often better for coersions, fwiw, as it won't throw on null/undefined (unless that's what you want).

Reply:
Other than above, the only other comment I have for this PR is if we are throwing any error if coercion is not possible on any type. 
(Is there any type that can't be coerced to string as such? What if we pass -1 or 1.23, does Service Bus throw meaningful errors as such?)

Reply:
> String(...) is often better for coersions

If we convert to `String`, then we can't pass it to methods that need `string`. We get type error.
To fix that, we will have to change all `string` to `String` in the code.

> as it won't throw on null/undefined (unless that's what you want).

In this case, I liked that `.toString()` throws the error. Beats us in having a specific checks for null/undefined

Reply:
`String(...)` returns a string not a String. `new String(...)` gives the string object. This is also true of other primitive types with object contructors, e.g. Number, Boolean, and Symbol.

Reply:
> if we are throwing any error if coercion is not possible on any type.
@ramya0820, In such cases, we are not throwing any error, but the `.toString()` will.

I believe `String()` will give a string no matter what the input.

Reply:
In conclusion, I can move to using `String()` in which case no error will be thrown right away regardless the input.